RevenueCatUI/CustomerCenter/Views/RestorePurchasesAlert.swift

```diff
@@ -54,17 +56,8 @@ struct RestorePurchasesAlert: ViewModifier {
                                     """),
                         primaryButton: .default(Text("Check past purchases"), action: {
                             Task {
-                                guard let customerInfo = try? await Purchases.shared.restorePurchases() else {
-                                    // todo: handle errors
-                                    self.setAlertType(.purchasesNotFound)
-                                    return
-                                }
-                                let hasEntitlements = customerInfo.entitlements.active.count > 0
-                                if hasEntitlements {
-                                    self.setAlertType(.purchasesRecovered)
-                                } else {
-                                    self.setAlertType(.purchasesNotFound)
-                                }
+                                let alertType = await self.customerCenterViewModel.performRestore() 
+                                self.setAlertType(alertType)
```

Comment:
I tried to move all this logic to the VM since I think it makes much more sense for it to be there. We could potentially create a different view model, but it's easier to pass the `CustomerCenterViewModel` around... Lmk what you think!

Reply:
fantastic